spec/features/recipes/show_spec.rb

```diff
@@ -0,0 +1,42 @@
+require "rails_helper"
+
+RSpec.describe "/recipes/:id, recipe show page", type: :feature do
+  describe "as a visitor to show page" do
+    before(:each) do
+      @recipe1 = Recipe.create!(name: "Spaghetti", complexity: 2, genre: "Italian")
+      @recipe2 = Recipe.create!(name: "Mac and Cheese", complexity: 1, genre: "American")
+
+      @ingredient1 = Ingredient.create!(name: "Ground Beef", cost: 2)
+      @ingredient2 = Ingredient.create!(name: "Salt", cost: 4)
+      @ingredient3 = Ingredient.create!(name: "Pasta", cost: 5)
+      @ingredient4 = Ingredient.create!(name: "Tomato", cost: 1)
+
+      RecipeIngredient.create!(recipe_id: @recipe1.id, ingredient_id: @ingredient1.id)
+      RecipeIngredient.create!(recipe_id: @recipe1.id, ingredient_id: @ingredient3.id)
+      RecipeIngredient.create!(recipe_id: @recipe1.id, ingredient_id: @ingredient4.id)
+    end
+
+    it "should see the specific recipe and it's attributes and ingredients needed" do
+      visit "/recipes/#{@recipe1.id}"
+
+      
+      expect(page).to have_content("Name: #{@recipe1.name}")
+      expect(page).to have_content("Complexity: #{@recipe1.complexity}")
+      expect(page).to have_content("Genre: #{@recipe1.genre}")
+
+      expect(page).to have_content("Ingredients:")
```

Comment:
Overall your feature tests are great, you have good object setup and show a step-by-step thought process which is awesome. 

It's important to remember, though, that whenever we write `expect(page).to have_content...` that capybara doesn't go line-by-line on the page too, it will search _the entire page_ for each line. 
So, just because we're expecting the page to have "Ingredients:" above the names of each ingredient, doesn't mean capybara will test that way. 
Instead, this is a perfect use case for a `within` block. If you had a `<div id="ingredients_list">` on your page, you could use a within block like so: 

```
within "#ingredients_list" do
   expect(page).to have_content(@ingredietn1.name)
   ...
end
```
And it would be a far more accurate & precise test of the view for this feature. Give it a try!